apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns-from-api.php

```diff
@@ -236,5 +241,51 @@ private function can_register_pattern( $pattern ) {
 
 		return true;
 	}
+
+	/**
+	 * Unregister all core patterns, then reregister core patterns in core Wordpress only,
+	 * that is those in wp-includes/block-patterns.php
+	 * Gutenberg adds new and overrides existing core patterns. We don't want these for now.
+	 */
+	private function reregister_core_patterns() {
+		if ( class_exists( 'WP_Block_Patterns_Registry' ) ) {
+			foreach ( \WP_Block_Patterns_Registry::get_instance()->get_all_registered() as $pattern ) {
+				// Gutenberg registers patterns with varying prefixes, but categorizes them using `core/*` in a blockTypes array.
+				// This will ensure we remove `query/*` blocks for example.
+				$pattern_block_type_or_name = ! empty( $pattern['blockTypes'][0] ) ? $pattern['blockTypes'][0] : $pattern['name'];
+				if ( 'core/' === substr( $pattern_block_type_or_name, 0, 5 ) ) {
+					unregister_block_pattern( $pattern['name'] );
+				}
+
+			}
+			if ( function_exists( '_register_core_block_patterns_and_categories' ) ) {
+				switch_to_locale( $this->utils->get_block_patterns_locale() );
```

Comment:
Do we need to switch back to the current locale again after doing this? And is there a performance penalty for switching locales? I imagine it's fine since we're only doing it once, but just thought I'd double check.

Reply:
It's probably safer if we do, thanks. I'll throw a `restore_previous_locale();` in there.

> And is there a performance penalty for switching locales?

Yes, I believe there is. Every time we switch locales we load the textdomain files. 

I'm not sure how much of a hit it'll be here, since we're switching to the site locale, which I guess we'd have to load elsewhere too anyway?